diff --git a/analytics/tests/test_stats_views.py b/analytics/tests/test_stats_views.py
index 68f6190f09..c600d4969b 100644
--- a/analytics/tests/test_stats_views.py
+++ b/analytics/tests/test_stats_views.py
@@ -17,7 +17,8 @@ class TestStatsEndpoint(ZulipTestCase):
     def test_stats(self) -> None:
         self.user = self.example_user("hamlet")
         self.login_user(self.user)
-        result = self.client_get("/stats")
+        with self.assert_database_query_count(6):
+            result = self.client_get("/stats")
         self.assertEqual(result.status_code, 200)
         # Check that we get something back
         self.assert_in_response("Zulip analytics for", result)
@@ -25,30 +26,36 @@ class TestStatsEndpoint(ZulipTestCase):
     def test_guest_user_cant_access_stats(self) -> None:
         self.user = self.example_user("polonius")
         self.login_user(self.user)
-        result = self.client_get("/stats")
+        with self.assert_database_query_count(4):
+            result = self.client_get("/stats")
         self.assert_json_error(result, "Not allowed for guest users", 400)
 
-        result = self.client_get("/json/analytics/chart_data")
+        with self.assert_database_query_count(2):
+            result = self.client_get("/json/analytics/chart_data")
         self.assert_json_error(result, "Not allowed for guest users", 400)
 
     def test_stats_for_realm(self) -> None:
         user = self.example_user("hamlet")
         self.login_user(user)
 
-        result = self.client_get("/stats/realm/zulip/")
+        with self.assert_database_query_count(4):
+            result = self.client_get("/stats/realm/zulip/")
         self.assertEqual(result.status_code, 302)
 
-        result = self.client_get("/stats/realm/not_existing_realm/")
+        with self.assert_database_query_count(4):
+            result = self.client_get("/stats/realm/not_existing_realm/")
         self.assertEqual(result.status_code, 302)
 
         user = self.example_user("hamlet")
         user.is_staff = True
         user.save(update_fields=["is_staff"])
 
-        result = self.client_get("/stats/realm/not_existing_realm/")
+        with self.assert_database_query_count(4):
+            result = self.client_get("/stats/realm/not_existing_realm/")
         self.assertEqual(result.status_code, 404)
 
-        result = self.client_get("/stats/realm/zulip/")
+        with self.assert_database_query_count(7):
+            result = self.client_get("/stats/realm/zulip/")
         self.assertEqual(result.status_code, 200)
         self.assert_in_response("Zulip analytics for", result)
 
@@ -56,14 +63,16 @@ class TestStatsEndpoint(ZulipTestCase):
         user = self.example_user("hamlet")
         self.login_user(user)
 
-        result = self.client_get("/stats/installation")
+        with self.assert_database_query_count(4):
+            result = self.client_get("/stats/installation")
         self.assertEqual(result.status_code, 302)
 
         user = self.example_user("hamlet")
         user.is_staff = True
         user.save(update_fields=["is_staff"])
 
-        result = self.client_get("/stats/installation")
+        with self.assert_database_query_count(5):
+            result = self.client_get("/stats/installation")
         self.assertEqual(result.status_code, 200)
         self.assert_in_response("Zulip analytics for", result)
 
@@ -137,7 +146,8 @@ class TestGetChartData(ZulipTestCase):
         self.insert_data(stat, [None], [])
         stat = COUNT_STATS["active_users_audit:is_bot:day"]
         self.insert_data(stat, ["false"], [])
-        result = self.client_get("/json/analytics/chart_data", {"chart_name": "number_of_humans"})
+        with self.assert_database_query_count(9):
+            result = self.client_get("/json/analytics/chart_data", {"chart_name": "number_of_humans"})
         data = self.assert_json_success(result)
         self.assertEqual(
             data,
@@ -158,9 +168,10 @@ class TestGetChartData(ZulipTestCase):
     def test_messages_sent_over_time(self) -> None:
         stat = COUNT_STATS["messages_sent:is_bot:hour"]
         self.insert_data(stat, ["true", "false"], ["false"])
-        result = self.client_get(
-            "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
-        )
+        with self.assert_database_query_count(6):
+            result = self.client_get(
+                "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
+            )
         data = self.assert_json_success(result)
         self.assertEqual(
             data,
@@ -180,9 +191,10 @@ class TestGetChartData(ZulipTestCase):
         self.insert_data(
             stat, ["public_stream", "private_message"], ["public_stream", "private_stream"]
         )
-        result = self.client_get(
-            "/json/analytics/chart_data", {"chart_name": "messages_sent_by_message_type"}
-        )
+        with self.assert_database_query_count(6):
+            result = self.client_get(
+                "/json/analytics/chart_data", {"chart_name": "messages_sent_by_message_type"}
+            )
         data = self.assert_json_success(result)
         self.assertEqual(
             data,
@@ -223,9 +235,10 @@ class TestGetChartData(ZulipTestCase):
             [str(client4.id), str(client3.id), str(client2.id)],
             [str(client3.id), str(client1.id)],
         )
-        result = self.client_get(
-            "/json/analytics/chart_data", {"chart_name": "messages_sent_by_client"}
-        )
+        with self.assert_database_query_count(7):
+            result = self.client_get(
+                "/json/analytics/chart_data", {"chart_name": "messages_sent_by_client"}
+            )
         data = self.assert_json_success(result)
         self.assertEqual(
             data,
@@ -247,9 +260,10 @@ class TestGetChartData(ZulipTestCase):
     def test_messages_read_over_time(self) -> None:
         stat = COUNT_STATS["messages_read::hour"]
         self.insert_data(stat, [None], [])
-        result = self.client_get(
-            "/json/analytics/chart_data", {"chart_name": "messages_read_over_time"}
-        )
+        with self.assert_database_query_count(6):
+            result = self.client_get(
+                "/json/analytics/chart_data", {"chart_name": "messages_read_over_time"}
+            )
         data = self.assert_json_success(result)
         self.assertEqual(
             data,
@@ -268,12 +282,13 @@ class TestGetChartData(ZulipTestCase):
         stat = COUNT_STATS["messages_in_stream:is_bot:day"]
         self.insert_data(stat, ["true", "false"], [])
 
-        result = self.client_get(
-            f"/json/analytics/chart_data/stream/{self.stream_id}",
-            {
-                "chart_name": "messages_sent_by_stream",
-            },
-        )
+        with self.assert_database_query_count(8):
+            result = self.client_get(
+                f"/json/analytics/chart_data/stream/{self.stream_id}",
+                {
+                    "chart_name": "messages_sent_by_stream",
+                },
+            )
         data = self.assert_json_success(result)
         self.assertEqual(
             data,
@@ -287,24 +302,26 @@ class TestGetChartData(ZulipTestCase):
             },
         )
 
-        result = self.api_get(
-            self.example_user("polonius"),
-            f"/api/v1/analytics/chart_data/stream/{self.stream_id}",
-            {
-                "chart_name": "messages_sent_by_stream",
-            },
-        )
+        with self.assert_database_query_count(4):
+            result = self.api_get(
+                self.example_user("polonius"),
+                f"/api/v1/analytics/chart_data/stream/{self.stream_id}",
+                {
+                    "chart_name": "messages_sent_by_stream",
+                },
+            )
         self.assert_json_error(result, "Not allowed for guest users")
 
         # Verify we correctly forbid access to stats of streams in other realms.
-        result = self.api_get(
-            self.mit_user("sipbtest"),
-            f"/api/v1/analytics/chart_data/stream/{self.stream_id}",
-            {
-                "chart_name": "messages_sent_by_stream",
-            },
-            subdomain="zephyr",
-        )
+        with self.assert_database_query_count(6):
+            result = self.api_get(
+                self.mit_user("sipbtest"),
+                f"/api/v1/analytics/chart_data/stream/{self.stream_id}",
+                {
+                    "chart_name": "messages_sent_by_stream",
+                },
+                subdomain="zephyr",
+            )
         self.assert_json_error(result, "Invalid channel ID")
 
     def test_include_empty_subgroups(self) -> None:
@@ -313,7 +330,8 @@ class TestGetChartData(ZulipTestCase):
             end_time=self.end_times_day[0],
             state=FillState.DONE,
         )
-        result = self.client_get("/json/analytics/chart_data", {"chart_name": "number_of_humans"})
+        with self.assert_database_query_count(9):
+            result = self.client_get("/json/analytics/chart_data", {"chart_name": "number_of_humans"})
         data = self.assert_json_success(result)
         self.assertEqual(data["everyone"], {"_1day": [0], "_15day": [0], "all_time": [0]})
         self.assertFalse("user" in data)
@@ -323,9 +341,10 @@ class TestGetChartData(ZulipTestCase):
             end_time=self.end_times_hour[0],
             state=FillState.DONE,
         )
-        result = self.client_get(
-            "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
-        )
+        with self.assert_database_query_count(6):
+            result = self.client_get(
+                "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
+            )
         data = self.assert_json_success(result)
         self.assertEqual(data["everyone"], {"human": [0], "bot": [0]})
         self.assertEqual(data["user"], {"human": [0], "bot": [0]})
@@ -335,9 +354,10 @@ class TestGetChartData(ZulipTestCase):
             end_time=self.end_times_day[0],
             state=FillState.DONE,
         )
-        result = self.client_get(
-            "/json/analytics/chart_data", {"chart_name": "messages_sent_by_message_type"}
-        )
+        with self.assert_database_query_count(6):
+            result = self.client_get(
+                "/json/analytics/chart_data", {"chart_name": "messages_sent_by_message_type"}
+            )
         data = self.assert_json_success(result)
         self.assertEqual(
             data["everyone"],
@@ -363,9 +383,10 @@ class TestGetChartData(ZulipTestCase):
             end_time=self.end_times_day[0],
             state=FillState.DONE,
         )
-        result = self.client_get(
-            "/json/analytics/chart_data", {"chart_name": "messages_sent_by_client"}
-        )
+        with self.assert_database_query_count(7):
+            result = self.client_get(
+                "/json/analytics/chart_data", {"chart_name": "messages_sent_by_client"}
+            )
         data = self.assert_json_success(result)
         self.assertEqual(data["everyone"], {})
         self.assertEqual(data["user"], {})
@@ -380,14 +401,15 @@ class TestGetChartData(ZulipTestCase):
         end_time_timestamps = [datetime_to_timestamp(dt) for dt in self.end_times_day]
 
         # valid start and end
-        result = self.client_get(
-            "/json/analytics/chart_data",
-            {
-                "chart_name": "number_of_humans",
-                "start": end_time_timestamps[1],
-                "end": end_time_timestamps[2],
-            },
-        )
+        with self.assert_database_query_count(9):
+            result = self.client_get(
+                "/json/analytics/chart_data",
+                {
+                    "chart_name": "number_of_humans",
+                    "start": end_time_timestamps[1],
+                    "end": end_time_timestamps[2],
+                },
+            )
         data = self.assert_json_success(result)
         self.assertEqual(data["end_times"], end_time_timestamps[1:3])
         self.assertEqual(
@@ -395,14 +417,15 @@ class TestGetChartData(ZulipTestCase):
         )
 
         # start later then end
-        result = self.client_get(
-            "/json/analytics/chart_data",
-            {
-                "chart_name": "number_of_humans",
-                "start": end_time_timestamps[2],
-                "end": end_time_timestamps[1],
-            },
-        )
+        with self.assert_database_query_count(3):
+            result = self.client_get(
+                "/json/analytics/chart_data",
+                {
+                    "chart_name": "number_of_humans",
+                    "start": end_time_timestamps[2],
+                    "end": end_time_timestamps[1],
+                },
+            )
         self.assert_json_error_contains(result, "Start time is later than")
 
     def test_min_length(self) -> None:
@@ -413,9 +436,10 @@ class TestGetChartData(ZulipTestCase):
         stat = COUNT_STATS["active_users_audit:is_bot:day"]
         self.insert_data(stat, ["false"], [])
         # test min_length is too short to change anything
-        result = self.client_get(
-            "/json/analytics/chart_data", {"chart_name": "number_of_humans", "min_length": 2}
-        )
+        with self.assert_database_query_count(9):
+            result = self.client_get(
+                "/json/analytics/chart_data", {"chart_name": "number_of_humans", "min_length": 2}
+            )
         data = self.assert_json_success(result)
         self.assertEqual(
             data["end_times"], [datetime_to_timestamp(dt) for dt in self.end_times_day]
@@ -425,9 +449,10 @@ class TestGetChartData(ZulipTestCase):
             {"_1day": self.data(100), "_15day": self.data(100), "all_time": self.data(100)},
         )
         # test min_length larger than filled data
-        result = self.client_get(
-            "/json/analytics/chart_data", {"chart_name": "number_of_humans", "min_length": 5}
-        )
+        with self.assert_database_query_count(9):
+            result = self.client_get(
+                "/json/analytics/chart_data", {"chart_name": "number_of_humans", "min_length": 5}
+            )
         data = self.assert_json_success(result)
         end_times = [
             ceiling_to_day(self.realm.date_created) + timedelta(days=i) for i in range(-1, 4)
@@ -443,7 +468,8 @@ class TestGetChartData(ZulipTestCase):
         )
 
     def test_non_existent_chart(self) -> None:
-        result = self.client_get("/json/analytics/chart_data", {"chart_name": "does_not_exist"})
+        with self.assert_database_query_count(3):
+            result = self.client_get("/json/analytics/chart_data", {"chart_name": "does_not_exist"})
         self.assert_json_error_contains(result, "Unknown chart name")
 
     def test_analytics_not_running(self) -> None:
@@ -454,9 +480,10 @@ class TestGetChartData(ZulipTestCase):
         realm.date_created = timezone_now() - timedelta(days=3)
         realm.save(update_fields=["date_created"])
         with self.assertLogs(level="WARNING") as m:
-            result = self.client_get(
-                "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
-            )
+            with self.assert_database_query_count(4):
+                result = self.client_get(
+                    "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
+                )
             self.assertEqual(
                 m.output,
                 [
@@ -469,9 +496,10 @@ class TestGetChartData(ZulipTestCase):
         realm.date_created = timezone_now() - timedelta(days=1, hours=2)
         realm.save(update_fields=["date_created"])
         with self.assertLogs(level="WARNING") as m:
-            result = self.client_get(
-                "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
-            )
+            with self.assert_database_query_count(4):
+                result = self.client_get(
+                    "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
+                )
             self.assertEqual(
                 m.output,
                 [
@@ -483,16 +511,18 @@ class TestGetChartData(ZulipTestCase):
 
         realm.date_created = timezone_now() - timedelta(days=1, minutes=10)
         realm.save(update_fields=["date_created"])
-        result = self.client_get(
-            "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
-        )
+        with self.assert_database_query_count(6):
+            result = self.client_get(
+                "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
+            )
         self.assert_json_success(result)
 
         realm.date_created = timezone_now() - timedelta(hours=10)
         realm.save(update_fields=["date_created"])
-        result = self.client_get(
-            "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
-        )
+        with self.assert_database_query_count(6):
+            result = self.client_get(
+                "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
+            )
         self.assert_json_success(result)
 
         end_time = timezone_now() - timedelta(days=5)
@@ -503,9 +533,10 @@ class TestGetChartData(ZulipTestCase):
         realm.date_created = timezone_now() - timedelta(days=3)
         realm.save(update_fields=["date_created"])
         with self.assertLogs(level="WARNING") as m:
-            result = self.client_get(
-                "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
-            )
+            with self.assert_database_query_count(4):
+                result = self.client_get(
+                    "/json/analytics/chart_data", {"chart_name": "messages_sent_over_time"}
+                )
             self.assertEqual(
                 m.output,
                 [
